src/charm.py

```diff
@@ -95,6 +143,107 @@ def _pebble_layer(self) -> ops.pebble.LayerDict:
             },
         }
 
+    @property
+    def _application_bind_address(self) -> str | None:
+        binding = self.model.get_binding("juju-info")
+        if not binding:
+            return None
+        if not binding.network:
+            return None
+        if not binding.network.bind_address:
+            return None
+        return str(binding.network.bind_address)
+
+    ## Status Checks ##
+    def _storages_attached(self) -> bool:
+        """Return if the storages are attached."""
+        return self.model.storages.get("config") and self.model.storages.get("database")  # type: ignore
+
+    def _gocert_available(self) -> bool:
+        """Return if the gocert server is reachable."""
+        try:
+            req = requests.get(
+                f"https://{self._application_bind_address}:{self.port}/status",
+                verify=f"{CHARM_PATH}/{CONFIG_MOUNT}/0/ca.pem",
+            )
+        except (requests.RequestException, OSError):
+            return False
+        if req.status_code != 200:
+            return False
+        return True
+
+    def _gocert_initialized(self) -> bool:
+        """Return if gocert is initialized."""
+        try:
+            req = requests.get(
+                f"https://{self._application_bind_address}:{self.port}/status",
+                verify=f"{CHARM_PATH}/{CONFIG_MOUNT}/0/ca.pem",
+            )
+        except (requests.RequestException, OSError):
+            return False
+        if req.status_code != 200:
+            return False
+        body = req.json()
+        return body.get("initialized", False)
```

Comment:
This feels like we'll soon want to move out gocert api call to their own class

Reply:
TLSENG-342